src/ucp/core/ucp_request.inl

```diff
@@ -63,6 +63,42 @@
     }
 
 
+static UCS_F_ALWAYS_INLINE ucp_recv_desc_t *
+ucp_recv_desc_get(ucp_worker_t *worker, void *data, void *desc, size_t length,
+                  uint16_t hdr_len, uint16_t flags)
+{
+    ucp_recv_desc_t *rdesc = (ucp_recv_desc_t *)data - 1;
+    if (rdesc == desc) {
+        rdesc->flags = flags | UCP_RECV_DESC_FLAG_UCT_DESC;
+    } else {
+        rdesc = ucs_mpool_get_inline(&worker->am_mp);
+        if (rdesc) {
```

Comment:
since we are creating the memory pool as infinite, i'd suggest (as optimization) to avoid checking whether rdesc!=NULL in the receive flow. we can just add an assertion.